### PR TITLE
FWI-4610 - /vv/add pod metadata support

### PR DESCRIPTION
**Pull request by vitorvezani**
This PR fixes #

## Checklist
* [X] I have signed the CLA
* [X] I have updated/added any relevant documentation

### What's the goal of this PR?
Expose `podMetadata` as part of the Workload

### CODE REVIEW

File: pkg/controller/pod.go [hunk 1]
```diff
@@ -18,26 +18,56 @@ import (
...
 	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 var podSpecFields = []string{"jobTemplate", "spec", "template"}
 
-// GetPodSpec looks inside arbitrary YAML for a PodSpec
-func GetPodSpec(obj map[string]interface{}) (*corev1.PodSpec, error) {
+// GetPodMetadataAndSpec looks inside arbitrary YAML for a PodSpec and it's metadata
+func GetPodMetadataAndSpec(obj map[string]any) (*metav1.ObjectMeta, *corev1.PodSpec, error) {
```
Comment (vitorvezani) [inline pkg/controller/pod.go:27]: I would get the metadata in another method, however, as I assume this is called a lot, that would double the processing time
So I took advantage and implemented along the spec search, this way, it should not affect performance that much

File: pkg/controller/pod_test.go [hunk 2]
```diff
@@ -19,27 +19,63 @@ import (
...
+	assert.NotNil(t, podSpec)
+	assert.Equal(t, 1, len(podMetadata.Labels))
+	assert.Equal(t, 1, len(podSpec.Containers))
+
+	podMetadata, podSpec, err = GetPodMetadataAndSpec(readFile(t, "./testdata/replication-controller.json"))
+	assert.NoError(t, err)
+	assert.NotNil(t, podSpec)
+	assert.Equal(t, 1, len(podMetadata.Labels))
+	assert.Equal(t, 1, len(podSpec.Containers))
+
```
Comment (jdesouza) [inline pkg/controller/pod_test.go:66]: Maybe adding the expected value for some of them would make the test more accurate

File: pkg/controller/pod.go (hunk 1 above)
Comment (rbren) [reply pkg/controller/pod.go:27]: Smart! Yeah this looks good

**Review by rbren — APPROVED** [review]
Nice!

File: pkg/controller/pod_test.go (hunk 2 above)
Comment (vitorvezani) [reply pkg/controller/pod_test.go:66]: done

**Pull request merged**